docs/sphinx/user/configuring_simeng.rst

```diff
@@ -145,20 +145,23 @@ The Branch-Prediction section contains those options to parameterise the branch
 
 The current options include:
 
+Type
+    The type of branch predictor that is used, the options are ``Generic``, and ``Perceptron``.  Both types of predictor use a branch target buffer with each entry containing a direction prediction mechanism and a target address.  The direction predictor used in ``Generic`` is a saturating counter, and in ``Perceptron`` it is a perceptron.
+
 BTB-Tag-Bits
     The number of bits used to denote an entry in the Branch Target Buffer (BTB). For example, a ``bits`` value of 12 could denote 4096 entries with the calculation 1 << ``bits``.
```

Comment:
I spoke with Jack about this section as I think it is a bit confusing. It would make sense to put it in this PR after agreement on wording. My suggested change is: "The number of bits used to represent a tag/index for the BTB. This therefore limits the size of the BTB to 2^bits entries. SimEng will use a BTB with the maximum number of entries based on this value. For example, a value of 12 means a BTB with 4096 entries will be used."

@jj16791 @FinnWilkinson @JosephMoore25 @ABenC377